rfcs/0013-deprecate-currency-factory.md

```diff
@@ -0,0 +1,345 @@
+* Native
+
+Acala has separate registration processes for each of these within their Asset 
+Registry pallet. Of interest to is are the foreign and native asset registration
+process.
+
+#### Foreign Assets
+
+Foreign assets are registered with a `MultiLocation` and `AssetMetadata`. The 
+`MultiLocation` then mapped to a `ForeignAssetId` that is determined by a nonce. 
+A mapping of `ForeignAssetId` to `MultiLocation` is also created so that either 
+can be used to look up the other. The provided metadata provides the assets 
+name, symbol, decimals, and minimum balance. The Metadata is indexed by the 
+`ForeignAssetId`.
+
+Foreign assets can then be updated by their `ForeignAssetId`, in which case
+both the assets `MultiLocation` and `AssetMetadata` can be updated.
+
+#### Native Assets
+
+Native assets are registered with a `CurrencyId` and `AssetMetadata`. The 
+`CurrencyId` maps to the `AssetMetadata` and this mapping is used to ensure that 
+the asset does not already exist.
+
+Native assets can then be updated by their `CurrencyId`, in which case
+the assets `AssetMetadata` can be updated.
+
+#### General Notes
+
+* The `AssetMetadatas` storage map is used for all types and is quarried by the 
+`AssetIds` `enum`.
+
+* The `AssetIds` type is also used for routing regular currency functions (i.e. 
+transfer, withdraw, deposit).
+
+* For non-erc20 assets, Acala uses 'orml-tokens' for currency functions
+
+#### Data Structures & Types
+
+Note: `MultiLocation` is supplied by XCM.
+
+```rust
+pub type ForeignAssetId = u16;
+```
+
+```rust
+pub struct AssetMetadata<Balance> {
+  pub name: Vec<u8>,
+	pub symbol: Vec<u8>,
+	pub decimals: u8,
+	pub minimal_balance: Balance,
+}
+```
+
+```rust
+pub enum AssetIds {
+	Erc20(EvmAddress),
+	StableAssetId(StableAssetPoolId),
+	ForeignAssetId(ForeignAssetId),
+	NativeAssetId(CurrencyId),
+}
+```
+
+```rust
+pub enum CurrencyId {
+	Token(TokenSymbol),
+	DexShare(DexShare, DexShare),
+	Erc20(EvmAddress),
+	StableAssetPoolToken(StableAssetPoolId),
+	LiquidCrowdloan(Lease),
+	ForeignAsset(ForeignAssetId),
+}
+```
+
+### Moonbeam
+
+Moonbeam has two asset classifications: 
+
+* Local
+* Foreign
+
+Moonbeam maintains separate instances of Parity's pallet-assets to track each of
+these asset classifications. Another pallet, pallet-asset-manager, is used to 
+route between the two.
+
+Local asset IDs are determined by the result of hashing a counter that tracks
+the number of local assets. While Foreign asset IDs are determined by the assets
+XCM Multilocation.
+
+#### Foreign
+
+Local assets are registered with a `ForeignAssetType`, `AssetRegisterarMetadata`, a `Balance` representing the ED, and a `bool` `is_sufficent`.
+
+## Requirements
+
+If we are to deprecate Currency Factory, we will need to ensure the requirements
+it fulfills are still met by some solution.
+
+* Permissioned asset creation can be done without collision
+
+* Asset metadata is available
+
+* Can registrar both local and foreign asset types
```

Comment:
Can we try to define what assets are considered foreign(assets created by other sovereign chains than the current executing chain) vs local(assets created by the chain that executes the logic). This will be helpful for anyone trying to understand how we handle assets.

Reply:
did not looked, but likely moonbeam uses Call filters to prevent any calls to extrinsic on foreign assets registry to prevent any minting or burning even via sudo.